xarray/coding/strings.py

```diff
@@ -191,8 +191,11 @@ def _numpy_char_to_bytes(arr):
     """Like netCDF4.chartostring, but faster and more flexible."""
     # based on: http://stackoverflow.com/a/10984878/809705
     arr = np.array(arr, copy=False, order="C")
-    dtype = "S" + str(arr.shape[-1])
-    return arr.view(dtype).reshape(arr.shape[:-1])
+    shape = arr.shape
+    dtype = "S" + str(shape[-1])
+    if arr.dtype.kind == "O":
+        arr = arr.astype("S1")
```

Comment:
Hmm. We should probably just fix this newly introduced issue over in h5netcdf first :)

Reply:
@shoyer Sure, I'll ping you over there.